lib/Sema/DerivedConformanceEquatableHashable.cpp

```diff
@@ -376,34 +1017,64 @@ deriveHashable_enum_hashValue(TypeChecker &tc, Decl *parentDecl,
   //     return index.hashValue
   //   }
   // }
+  //
+  // enum SomeEnumWithAssociatedValues {
+  //   case A, B(Int), C(String, Int)
+  //   @derived var hashValue: Int {
+  //     var result: Int
+  //     switch self {
+  //     case A:
+  //       result = _mixInt(0)
```

Comment:
I think we could devise a more efficient function to use here than `_mixInt`. Its purpose is to increase the variability of hash values before using them with hashed collections, so calling it multiple times does more work than is necessary. For example, each call incorporates the per-execution seed, which isn't necessary at the instance level if it's happening at the type level.

Additionally, an enum case with no associated values or a struct with only a single property shouldn't need mixing at all. That is, this could just be `result = 0`, while the struct implementation could start with `var result = x.hashValue`.